Examples/PcapSplitter/IPPortSplitters.h

```diff
@@ -233,6 +233,32 @@ class IPPortSplitter : public ValueBasedSplitter
 		return "miscellaneous";
 	}
 
+	/**
+	 * An auxiliary method to indicate whether an IPv4/IPv6 source address is multicast or not
+	 */
+	bool isSrcIPMulticast(pcpp::Packet& packet)
+	{
+		if (packet.isPacketOfType(pcpp::IPv4))
+			return packet.getLayerOfType<pcpp::IPv4Layer>()->getSrcIPv4Address().isMulticast();
+		else if (packet.isPacketOfType(pcpp::IPv6))
+			return packet.getLayerOfType<pcpp::IPv6Layer>()->getSrcIPv6Address().isMulticast();
+		else
+			return false;

```

Comment:
if we add `isMulticast()` to `IPAddress` we can simplify this method and `isDstIPMulticast()`